source/patterns/components/video/video.js

```diff
@@ -0,0 +1,42 @@
+// A simple script to pause videos when they aren't visible in the viewport.
+// @todo Add logic to deal with videos added to the DOM via ajax or other.
+// @todo Deal with videos in iframes. See https://stackoverflow.com/a/15251023 and https://stackoverflow.com/q/12336031
+(function (window, document) {
+
+  const cu_video_elements = document.querySelectorAll('.cu-video');
+
+  const inViewport = function(el) {
+    let bounding = el.getBoundingClientRect();
+    return (bounding.top < window.innerHeight) && (bounding.bottom > 0);
```

Comment:
Can you discuss this with @daviddemello? He suggested we pause it if it's partially off-screen, rather than totally off-screen. He mentioned that seeing just part of the video as you start scrolling is distracting. Something like `bounding.bottom > bounding.height / 2` would pause it when half the video is no longer visible, for example.

Reply:
Excellent suggestion. I added your bit about the top of viewport boundary and I also did something similar for videos scrolled below the viewport. That's in commit 6dc5ccc.